### PR TITLE
Delete series fixes

### PR DESCRIPTION
**Pull request by jwilder**
###### Required for all non-trivial PRs
- [x] Rebased/mergable
- [x] Tests pass

This fixes a regression introduced in #6501 and fixes a few bugs in deleting series.  The specific issues fixes:
* Empty tombstone files could be written and would fail to be loaded if you dropped a series that didn't actually exist in a TSM file.
* Fixed some logging prefix that were missing a space
* Optimized DropSeries to be more efficient since it's called more frequently now
* Reverted filtering out index entries in the TSM block iterator (added in #6501) since it was fixing the wrong problem.
* Optimized loading and applying large numbers of tombstones
* Fixed Series not getting tagged with the shard that owns them we loading shards

### CODE REVIEW

**Comment by jwilder** [comment]
@e-dard

File: tsdb/engine/tsm1/reader.go [hunk 1]
```diff
@@ -127,13 +127,10 @@ func (b *BlockIterator) Next() bool {
 		}
 	}
 
-	for b.n-b.i > 0 {
+	if b.n-b.i > 0 {
```
Comment (benbjohnson) [inline tsdb/engine/tsm1/reader.go]: Why is this changed from a loop to a conditional?

File: tsdb/meta.go [hunk 2]
```diff
@@ -1815,6 +1802,21 @@ func (s stringSet) intersect(o stringSet) stringSet {
...
+	// binary search for v
+	i := sort.Search(len(a), func(i int) bool { return a[i] >= v })
+	if i < len(a) && a[i] == v {
+		// we found it, so shift the right half down one, overwriting v's
+		// position.
+		copy(a[i:], a[i+1:])
+		return a[:len(a)-1]
+	} else {
+		return a
+	}
```
Comment (benbjohnson) [inline tsdb/meta.go]: Shortcutting the `return a` removes some indentation:

```go
func filter(a []uint64, v uint64) []uint64 {
	// binary search for v
	i := sort.Search(len(a), func(i int) bool { return a[i] >= v })
	if i >= len(a) || a[i] != v {
		return a
	}

	// we found it, so shift the right half down one, overwriting v's position.
	copy(a[i:], a[i+1:])
	return a[:len(a)-1]
}
```

**Comment by benbjohnson** [comment]
Minor nits but otherwise lgtm. :+1:

File: tsdb/engine/tsm1/reader.go (hunk 1 above)
Comment (jwilder) [inline tsdb/engine/tsm1/reader.go]: It was the wrong fix I added in #6501.  I'm reverting it here because the PR has the right fix down in the the tombstones.
